### ISSUE TITLE
sync gets stuck at the end after large copy

### ISSUE THREAD

**Issue by gerph** [rclone/rclone#4676]
#### What is the problem you are having with rclone?

I'm running a sync - usually between two remotes - and sometimes the sync does a lot of copying, and maybe some deleting and then gets stuck at the end doing nothing with (seemingly) no network traffic. At least the traffic monitoring is showing low-Kilobytes, whereas in the earlier stages it was in the 100s of KB/s or at times MB/s.

#### What is your rclone version (output from `rclone version`)

```
charles@laputa ~> rclone version
rclone v1.54.0-beta.4811.e3a5bb9b4
- os/arch: darwin/amd64
- go version: go1.15.2
```

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)

macOS 10.14.6

####  Which cloud storage system are you using? (eg Google Drive)

Jottacloud, and Hubic.

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)

Example:

```rclone -v sync --checkers 15 --transfers 15 hubic-crypt:archives/2019-02/home/ --exclude expect/** --exclude charles/projects/icons/backup/icons.db --exclude charles/music/** jottacloud-crypt:archives/2019-02/home/```

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)

I don't have logs with -vv I'm afraid, but I've got some with -v.

```
rclone -v sync --checkers 15 --transfers 15 hubic-crypt:archives/2019-02/home/ --exclude expect/** --exclude charles/projects/icons/backup/icons.db --exclude charles/music/** jottacloud-crypt:archives/2019-02/home/
2020/10/12 23:58:43 INFO  : 
Transferred:   	         0 / 0 Bytes, -, 0 Bytes/s, ETA -
Checks:              1355 / 1542, 88%
Elapsed time:       1m9.7s
Checking:
 *      charles/ansible.gerph.old/.git/logs/HEAD: checking
 *   charles/ansible.gerph.old/fork/results3.txt: checking
 *   charles/ansible.gerph.old/fork/results5.txt: checking
 *   charles/ansible.gerph.old/fork/results7.txt: checking
 *          charles/datasets/election/carto.json: checking
 *         charles/datasets/election/concan.json: checking
 *     charles/datasets/election/conresults.json: checking
 *           charles/datasets/election/cons.json: checking
 * charles/datasets/elect…nstituencies.topo.json: checking
 *          charles/datasets/election/fetcher.sh: checking
 *        charles/datasets/election/parties.json: checking
 *      charles/datasets/election/ukresults.json: checking
 *      charles/datasets/election/voteshare.json: checking
 * charles/Reference/ARM/…/DDI0151C_920T_TRM.pdf: checking
 *      charles/ansible.gerph.old/group_vars/all: checking

... spin on for a bit ...

Transferred:   	         0 / 0 Bytes, -, 0 Bytes/s, ETA -
Checks:            177742 / 177757, 100%
Deleted:             9659
Elapsed time:    1h32m9.7s
Checking:
 * charles/Reference/book…'s Word Puzzle (J).nes: checking
 * charles/.cpanm/work/15…3/devel/grep-pod-gt.pl: checking
 * charles/external/Image…ore_7_Q16HDRI_la-pcd.o: checking
 * charles/riscos/.git/mo…ooks/pre-commit.sample: checking
 * charles/riscos/.git/mo…ox/FontMenu/FETCH_HEAD: checking
 * charles/no-backup/!Wor…pstar/wimpc103.zip,ddc: checking
 * charles/no-backup/!Wor…Music/d1mus/D_E1M4,308: checking
 * charles/projects/robui…service/.gitlab-ci.yml: checking
 * charles/riscos/Sources…ils/CMunge/h/writefile: checking
 *        charles/external/graphviz/FEATURE/sfio: checking
 * charles/external/Image…_7_Q16HDRI_la-ycbcr.lo: checking
 * charles/.cpanm/work/15…erl-Critic-1.126/Build: checking
 * charles/external/Image…nimate-private_8h.html: checking
 *        charles/riscos/Export/Lib/CLX/h/disass: checking

2020/10/13 01:29:44 INFO  : charles/riscos/Export/Lib/CLX/h/disass: Deleted
2020/10/13 01:29:44 INFO  : charles/projects/justin/Sources/Applications/Clippy/VersionBas,ffb: Deleted
2020/10/13 01:29:44 INFO  : charles/projects/justin/Sources/Games/Doom/Utils/Slige/Resources/!MakeDiffs,fd7: Deleted

... skip many deletions ...

2020/10/13 01:31:43 INFO  : 
Transferred:   	         0 / 0 Bytes, -, 0 Bytes/s, ETA -
Checks:            179018 / 179018, 100%
Deleted:            10920
Elapsed time:    1h34m9.7s

... skip the next 12 hours of the same message ...

2020/10/13 13:46:42 INFO  : 
Transferred:   	         0 / 0 Bytes, -, 0 Bytes/s, ETA -
Checks:            179018 / 179018, 100%
Deleted:            10920
Elapsed time:   13h49m9.7s
```

It's still running, but it's not at all clear what it's doing.

Another example, using a slightly different directory:

```rclone -v sync --checkers 15 --transfers 15 hubic-crypt:archives/2019-03/home/ --exclude expect/** --exclude charles/projects/icons/backup/icons.db --exclude charles/music/** jottacloud-crypt:archives/2019-03/home/
2020/10/12 16:07:37 INFO  : 
Transferred:   	         0 / 0 Bytes, -, 0 Bytes/s, ETA -
Checks:              1259 / 1397, 90%
Elapsed time:       1m9.6s
Checking:
 * charles/ansible.gerph.…d/faillog/kanboard.txt: checking
 * charles/ansible.gerph.…faillog/local-gems.txt: checking
 *  charles/ansible.gerph.old/faillog/locate.txt: checking
 * charles/ansible.gerph.…/faillog/mediawiki.txt: checking
 * charles/ansible.gerph.…illog/munin-server.txt: checking
 * charles/ansible.gerph.…l-server-databases.txt: checking
 * charles/ansible.gerph.…aillog/nagios-nrpe.txt: checking
 * charles/ansible.gerph.…llog/nagios-server.txt: checking
 * charles/ansible.gerph.…faillog/nameserver.txt: checking
 * charles/ansible.gerph.…faillog/nfs-client.txt: checking
 *  charles/ansible.gerph.old/faillog/oclint.txt: checking
 * charles/ansible.gerph.…oefenweb.wordpress.txt: checking
 * charles/ansible.gerph.…log/openvpn-server.txt: checking
 * charles/ansible.gerph.…/faillog/orangehrm.txt: checking
 * charles/ansible.gerph.…d/faillog/owncloud.txt: checking

... skip on to where it stops reporting checks ...

2020/10/12 17:45:37 INFO  : 
Transferred:   	         0 / 0 Bytes, -, 0 Bytes/s, ETA -
Checks:            193438 / 193438, 100%
Elapsed time:    1h39m9.6s

... skip on to where it completes ...


2020/10/12 23:57:33 INFO  : There was nothing to transfer
2020/10/12 23:57:33 INFO  : 
Transferred:   	         0 / 0 Bytes, -, 0 Bytes/s, ETA -
Checks:            193438 / 193438, 100%
Elapsed time:    7h51m6.5s
```

**Comment by ncw** [comment]
What would be really helpful is if you could `kill -QUIT` rclone when it gets stuck. I think `killall -QUIT rclone` should work if issued from another terminal window (sorry not a mac user!).

This should generate a traceback - if you could post the entire thing here then I can work out what is going on.

Thanks.

**Comment by gerph** [comment]
```

2020/10/13 15:36:41 INFO  : 
Transferred:   	         0 / 0 Bytes, -, 0 Bytes/s, ETA -
Checks:            179018 / 179018, 100%
Deleted:            10920
Elapsed time:   15h39m9.7s

SIGQUIT: quit
PC=0x7fff70c88866 m=0 sigcode=0

goroutine 0 [idle]:
runtime.pthread_cond_wait(0x643a540, 0x643a500, 0x0)
	runtime/sys_darwin.go:414 +0x39
runtime.semasleep(0xffffffffffffffff, 0x24)
	runtime/os_darwin.go:63 +0x8d
runtime.notesleep(0x643a308)
	runtime/lock_sema.go:181 +0xe7
runtime.stopm()
	runtime/proc.go:1910 +0xc5
runtime.exitsyscall0(0xc00009be00)
	runtime/proc.go:3401 +0x11d
runtime.mcall(0x40704f6)
	runtime/asm_amd64.s:318 +0x5b

goroutine 1 [sleep]:
time.Sleep(0x3b9aca00)
	runtime/time.go:188 +0xbf
github.com/rclone/rclone/lib/oauthutil.(*TokenSource).Token(0xc00069a0c0, 0x0, 0x0, 0x0)
	github.com/rclone/rclone/lib/oauthutil/oauthutil.go:234 +0xac
golang.org/x/oauth2.(*reuseTokenSource).Token(0xc001934260, 0x0, 0x0, 0x0)
	golang.org/x/oauth2@v0.0.0-20200107190931-bf48bf16ab8d/oauth2.go:304 +0xaf
golang.org/x/oauth2.(*Transport).RoundTrip(0xc001934280, 0xc00087e000, 0x0, 0x0, 0x0)
	golang.org/x/oauth2@v0.0.0-20200107190931-bf48bf16ab8d/transport.go:45 +0x9e
net/http.send(0xc00087e000, 0x597cee0, 0xc001934280, 0x0, 0x0, 0x0, 0xc0000b0460, 0x400, 0x1, 0x0)
	net/http/client.go:252 +0x453
net/http.(*Client).send(0xc001900300, 0xc00087e000, 0x0, 0x0, 0x0, 0xc0000b0460, 0x0, 0x1, 0x0)
	net/http/client.go:176 +0xff
net/http.(*Client).do(0xc001900300, 0xc00087e000, 0x0, 0x0, 0x0)
	net/http/client.go:718 +0x45f
net/http.(*Client).Do(...)
	net/http/client.go:586
github.com/rclone/rclone/lib/rest.(*Client).Call(0xc001906000, 0x59b5a80, 0xc0019223c0, 0xc00014cb40, 0x0, 0x0, 0x0)
	github.com/rclone/rclone/lib/rest/rest.go:280 +0x946
github.com/rclone/rclone/lib/rest.(*Client).callCodec(0xc001906000, 0x59b5a80, 0xc0019223c0, 0xc00014cb40, 0x0, 0x0, 0x5314540, 0xc00083c480, 0x577d208, 0x577f3d8, ...)
	github.com/rclone/rclone/lib/rest/rest.go:464 +0x212
github.com/rclone/rclone/lib/rest.(*Client).CallXML(...)
	github.com/rclone/rclone/lib/rest/rest.go:424
github.com/rclone/rclone/backend/jottacloud.(*Fs).List.func1(0xc00190c1d8, 0x8, 0xc000c0f1e0)
	github.com/rclone/rclone/backend/jottacloud/jottacloud.go:832 +0xbd
github.com/rclone/rclone/fs.pacerInvoker(0x1, 0xa, 0xc000812bc0, 0xc000812bc0, 0x33f81698, 0x0)
	github.com/rclone/rclone/fs/fs.go:1477 +0x3c
github.com/rclone/rclone/lib/pacer.(*Pacer).call(0xc00069a120, 0xc000812bc0, 0xa, 0x40, 0x55daa80)
	github.com/rclone/rclone/lib/pacer/pacer.go:197 +0x90
github.com/rclone/rclone/lib/pacer.(*Pacer).Call(0xc00069a120, 0xc000812bc0, 0x10d, 0xc000c0f1e0)
	github.com/rclone/rclone/lib/pacer/pacer.go:216 +0x6a
github.com/rclone/rclone/backend/jottacloud.(*Fs).List(0xc001956210, 0x59b5a80, 0xc0019223c0, 0xc002da8120, 0x10d, 0xc000000001, 0x10000c001b4a440, 0x0, 0x10d, 0xc002da8120)
	github.com/rclone/rclone/backend/jottacloud/jottacloud.go:831 +0x1ff
github.com/rclone/rclone/backend/jottacloud.(*Fs).purgeCheck(0xc001956210, 0x59b5a80, 0xc0019223c0, 0xc002da8120, 0x10d, 0x1, 0x90, 0xc001771680)
	github.com/rclone/rclone/backend/jottacloud/jottacloud.go:1024 +0x1c5
github.com/rclone/rclone/backend/jottacloud.(*Fs).Rmdir(0xc001956210, 0x59b5a80, 0xc0019223c0, 0xc002da8120, 0x10d, 0x1, 0xc0017a1cc0)
	github.com/rclone/rclone/backend/jottacloud/jottacloud.go:1064 +0x58
github.com/rclone/rclone/backend/crypt.(*Fs).Rmdir(0xc000536000, 0x59b5a80, 0xc0019223c0, 0xc001a33950, 0x41, 0x10, 0xc00234e200)
	github.com/rclone/rclone/backend/crypt/crypt.go:430 +0x6d
github.com/rclone/rclone/fs/operations.TryRmdir(0x59b5a80, 0xc0019223c0, 0x59cbd20, 0xc000536000, 0xc001a33950, 0x41, 0x597bdc0, 0xc00234e240)
	github.com/rclone/rclone/fs/operations/operations.go:924 +0xf0
github.com/rclone/rclone/fs/sync.deleteEmptyDirectories(0x59b5a80, 0xc0019223c0, 0x59cbd20, 0xc000536000, 0xc0019007e0, 0xc00069a660, 0xc00069a600)
	github.com/rclone/rclone/fs/sync/sync.go:549 +0x3d1
github.com/rclone/rclone/fs/sync.(*syncCopyMove).run(0xc00000c3c0, 0xc00013e008, 0x59cbd20)
	github.com/rclone/rclone/fs/sync/sync.go:875 +0x57e
github.com/rclone/rclone/fs/sync.runSyncCopyMove(0x59b5ac0, 0xc00013e008, 0x59cbd20, 0xc000536000, 0x59cbd20, 0xc0018146e0, 0xc000000003, 0x577eb60, 0xc00069a540)
	github.com/rclone/rclone/fs/sync/sync.go:1065 +0x27c
github.com/rclone/rclone/fs/sync.Sync(...)
	github.com/rclone/rclone/fs/sync/sync.go:1070
github.com/rclone/rclone/cmd/sync.glob..func1.1(0xc0019349a0, 0x51e7a602)
	github.com/rclone/rclone/cmd/sync/sync.go:55 +0x8f
github.com/rclone/rclone/cmd.Run(0xc0001c0101, 0x6428420, 0xc0001f3d40)
	github.com/rclone/rclone/cmd/cmd.go:248 +0xc3
github.com/rclone/rclone/cmd/sync.glob..func1(0x6428420, 0xc0001cc410, 0x2, 0xd)
	github.com/rclone/rclone/cmd/sync/sync.go:53 +0x12a
github.com/spf13/cobra.(*Command).execute(0x6428420, 0xc0001cc340, 0xd, 0xd, 0x6428420, 0xc0001cc340)
	github.com/spf13/cobra@v1.0.0/command.go:846 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0x641d1a0, 0x5f84df5d, 0x6438ce0, 0xc000118058)
	github.com/spf13/cobra@v1.0.0/command.go:950 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.0.0/command.go:887
github.com/rclone/rclone/cmd.Main()
	github.com/rclone/rclone/cmd/cmd.go:538 +0x9a
main.main()
	github.com/rclone/rclone/rclone.go:14 +0x25

goroutine 6 [select]:
go.opencensus.io/stats/view.(*worker).start(0xc000304700)
	go.opencensus.io@v0.22.4/stats/view/worker.go:276 +0x105
created by go.opencensus.io/stats/view.init.0
	go.opencensus.io@v0.22.4/stats/view/worker.go:34 +0x68

goroutine 74 [chan receive, 15 minutes]:
github.com/rclone/rclone/lib/oauthutil.(*Renew).renewOnExpiry(0xc0019003f0)
	github.com/rclone/rclone/lib/oauthutil/renew.go:39 +0x68
created by github.com/rclone/rclone/lib/oauthutil.NewRenew
	github.com/rclone/rclone/lib/oauthutil/renew.go:28 +0x85

goroutine 98 [select]:
github.com/rclone/rclone/cmd.StartStats.func2(0xc001908a10, 0xc00190c900)
	github.com/rclone/rclone/cmd/cmd.go:357 +0x11d
created by github.com/rclone/rclone/cmd.StartStats
	github.com/rclone/rclone/cmd/cmd.go:353 +0xb1

goroutine 100 [syscall, 939 minutes]:
os/signal.signal_recv(0x5781790)
	runtime/sigqueue.go:144 +0x9d
os/signal.loop()
	os/signal/signal_unix.go:23 +0x25
created by os/signal.Notify.func1.1
	os/signal/signal.go:150 +0x45

goroutine 101 [chan receive, 939 minutes]:
github.com/rclone/rclone/cmd.SigInfoHandler.func1(0xc00069a540)
	github.com/rclone/rclone/cmd/siginfo_darwin.go:19 +0xb7
created by github.com/rclone/rclone/cmd.SigInfoHandler
	github.com/rclone/rclone/cmd/siginfo_darwin.go:18 +0xab

goroutine 147 [chan receive, 939 minutes]:
github.com/rclone/rclone/fs/march.(*March).Run.func2(0xc0018985a0, 0xc00069a780, 0xc001908a60)
	github.com/rclone/rclone/fs/march/march.go:199 +0x4e
created by github.com/rclone/rclone/fs/march.(*March).Run
	github.com/rclone/rclone/fs/march/march.go:197 +0x306

goroutine 1442291 [select]:
net/http.(*persistConn).writeLoop(0xc0049025a0)
	net/http/transport.go:2340 +0x11c
created by net/http.(*Transport).dialConn
	net/http/transport.go:1709 +0xcdc

goroutine 1442290 [IO wait]:
internal/poll.runtime_pollWait(0x33c46a68, 0x72, 0x5981ca0)
	runtime/netpoll.go:220 +0x55
internal/poll.(*pollDesc).wait(0xc000305e98, 0x72, 0xc000f4e000, 0x16e3, 0x16e3)
	internal/poll/fd_poll_runtime.go:87 +0x45
internal/poll.(*pollDesc).waitRead(...)
	internal/poll/fd_poll_runtime.go:92
internal/poll.(*FD).Read(0xc000305e80, 0xc000f4e000, 0x16e3, 0x16e3, 0x0, 0x0, 0x0)
	internal/poll/fd_unix.go:159 +0x1b1
net.(*netFD).Read(0xc000305e80, 0xc000f4e000, 0x16e3, 0x16e3, 0x16d6, 0x7e841f4206e21fb5, 0xd974d351b4c68c90)
	net/fd_posix.go:55 +0x4f
net.(*conn).Read(0xc0000106f8, 0xc000f4e000, 0x16e3, 0x16e3, 0x0, 0x0, 0x0)
	net/net.go:182 +0x8e
github.com/rclone/rclone/fs/fshttp.(*timeoutConn).readOrWrite(0xc000f6cd00, 0xc00066f580, 0xc000f4e000, 0x16e3, 0x16e3, 0x203000, 0x4280dbb, 0xc002148860)
	github.com/rclone/rclone/fs/fshttp/http.go:75 +0x48
github.com/rclone/rclone/fs/fshttp.(*timeoutConn).Read(0xc000f6cd00, 0xc000f4e000, 0x16e3, 0x16e3, 0xc0021488a8, 0xd, 0x10)
	github.com/rclone/rclone/fs/fshttp/http.go:87 +0x8a
crypto/tls.(*atLeastReader).Read(0xc00234e980, 0xc000f4e000, 0x16e3, 0x16e3, 0x163, 0x16d6, 0xc00066f668)
	crypto/tls/conn.go:779 +0x62
bytes.(*Buffer).ReadFrom(0xc002148980, 0x597ae00, 0xc00234e980, 0x400d6e5, 0x5447a20, 0x55b3fc0)
	bytes/buffer.go:204 +0xb1
crypto/tls.(*Conn).readFromUntil(0xc002148700, 0xba5a518, 0xc000f6cd00, 0x5, 0xc000f6cd00, 0x100)
	crypto/tls/conn.go:801 +0xf3
crypto/tls.(*Conn).readRecordOrCCS(0xc002148700, 0x0, 0x0, 0x0)
	crypto/tls/conn.go:608 +0x115
crypto/tls.(*Conn).readRecord(...)
	crypto/tls/conn.go:576
crypto/tls.(*Conn).Read(0xc002148700, 0xc000701000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	crypto/tls/conn.go:1252 +0x15f
net/http.(*persistConn).Read(0xc0049025a0, 0xc000701000, 0x1000, 0x1000, 0xc003d06660, 0xc00066fc58, 0x4007895)
	net/http/transport.go:1887 +0x77
bufio.(*Reader).fill(0xc002c88060)
	bufio/bufio.go:101 +0x105
bufio.(*Reader).Peek(0xc002c88060, 0x1, 0x0, 0x0, 0x1, 0x0, 0xc0026d1860)
	bufio/bufio.go:139 +0x4f
net/http.(*persistConn).readLoop(0xc0049025a0)
	net/http/transport.go:2040 +0x1a8
created by net/http.(*Transport).dialConn
	net/http/transport.go:1708 +0xcb7

rax    0x104
rbx    0x2
rcx    0x7ffeefbfeaa8
rdx    0x3d12900
rdi    0x643a540
rsi    0x3d12e0103d12f00
rbp    0x7ffeefbfeb30
rsp    0x7ffeefbfeaa8
r8     0x0
r9     0xa0
r10    0x0
r11    0x202
r12    0x643a540
r13    0x16
r14    0x3d12e0103d12f00
r15    0xbd165c0
rip    0x7fff70c88866
rflags 0x203
cs     0x7
fs     0x0
gs     0x0

2020/10/13 15:37:19 Failed to create file system for "jottacloud-crypt:archives/2019-01/home/": failed to make remote "jottacloud:/data/" to wrap: couldn't get customer info: Get "https://api.jottacloud.com/account/v1/customer": couldn't fetch token - maybe it has expired? - refresh with "rclone config reconnect jottacloud:": oauth2: cannot fetch token: 400 Bad Request
Response: {"error":"invalid_grant","error_description":"Stale token"}
```

The final log entry was from the /next/ rclone operation that had been run I was running them in a loop to handle 2019-03, 2019-02, and 2019-01. 2019-02 was the one that was stuck, and which I issued the -QUIT to. It exited and the loop contined on to the 01, where it reported that error. I don't know if that's a relevant error though. In the other runs I've done even when it completed the token was still valid.

However, trying it now with an `lsd` I do get that error:

```
rclone -v lsd --checkers 15 --transfers 25 jottacloud-crypt:archives/2019-03/home/charles/music
2020/10/13 15:42:27 Failed to create file system for "jottacloud-crypt:archives/2019-03/home/charles/music": failed to make remote "jottacloud:/data/" to wrap: couldn't get customer info: Get "https://api.jottacloud.com/account/v1/customer": couldn't fetch token - maybe it has expired? - refresh with "rclone config reconnect jottacloud:": oauth2: cannot fetch token: 400 Bad Request
Response: {"error":"invalid_grant","error_description":"Stale token"}
```

I hope something in that is useful to you - I'll go get a new token and try the sync again :-)

**Comment by ncw** [comment]
In the traceback, rclone is trying to refresh the token. Everything else looks absolutely normal.

Can you try again and see if you can replicate the problem and post the traceback again.

It would be worth running with -vv and uploading the log too - rclone will be printing stuff about renewing the token...

Thanks

**Comment by gerph** [comment]
Interesting; the next transfer I go through, I'll add a -vv option and then I may have more info for you - I'm transferring backups from the hubic provider to jottacloud, so I've got quite a few more transfers still to do.

I presume -vv will include data I'll need to obscure otherwise it'll give away secrets? Is there an easy way to spot what needs obscuring?

**Comment by gerph** [comment]
I ran a transfer with -vv and I can now see the long period where the number of files reported is unchanging, and can see that the traffic is very low, BUT with -vv in the list I can now see what it's doing.

```
Transferred:   	         0 / 0 Bytes, -, 0 Bytes/s, ETA -
Checks:            168098 / 168098, 100%
Elapsed time:    3h38m5.8s

2020/10/13 21:20:06 DEBUG : charles/.indent.pro: Removing directory
2020/10/13 21:20:10 DEBUG : charles/.cpanm/work/1584575198.11594: Removing directory
2020/10/13 21:20:13 DEBUG : charles/.cpanm/work/1584575178.10998: Removing directory
2020/10/13 21:20:16 DEBUG : charles/.cpanm/work/1584574184.8584: Removing directory
2020/10/13 21:20:18 DEBUG : charles/.cpanm/work/1584573965.8022: Removing directory
2020/10/13 21:20:22 DEBUG : charles/.cpanm/work/1584573940.7434: Removing directory
2020/10/13 21:20:25 DEBUG : charles/.cpanm/work/1584572852.4411: Removing directory
2020/10/13 21:20:27 DEBUG : charles/.cpanm/work/1584572828.3873: Removing directory
... and so on ...
```

As you can see, it /is/ doing something... once every 2 to 3 seconds it's deleting a directory. And that's probably why it's taking so long, and appears to be doing nothing. There doesn't seem to be a configuration for the number of deleter jobs (--transfers and --checkers, I have both set to 15), and maybe it's using the --transfers flag but it really is just taking a long time and every 3 seconds one of those requests completes.

It'd be nice to have some feedback that it's actually achieving something during that period when it's deleting directories, though.

**Comment by gerph** [comment]
I've had the -vv switch enabled for the past few days doing the copy, and it's looking like it is always doing something during that seemingly idle period at the end - it's just deleting the directories, as noticed in the prior comment. Because there's no feedback in the statistics of what it's doing, that looked like it was stuck when it really was doing things. In my case I think I had exacerbated the problem because of how I was doing the copy.

I would `copy` from hubic->jottacloud one set of backups for a month. This would download the content to my machine and then upload to jottacloud, so it's pretty tedious. Then I would do a jottacloud->jottacloud `copy` to copy that backup to the previous months. This would do a server side copy, so all those files would be copied really quickly and there would be no need for anything other than the control operations. Then I would do a `sync` from hubc->jottacloud the actual backups for those months.

Because I'm going backwards in time, this resulted in (mostly) deleting files that weren't present and restoring them to the older versions. The further back in time I went with my copy of the monthly backups, the greater the differences and the greater the number of files and directories to be deleted. For one of the backups, there were 310K files checked, 120K files deleted, and 43K directories deleted - and those directory deletions took a long time for each individual one, as mentioned.

```
2020/10/16 13:35:00 DEBUG : Encrypted drive 'jottacloud-crypt:archives/2018-11/home/': failed to delete 2 directories
2020/10/16 13:35:00 DEBUG : Encrypted drive 'jottacloud-crypt:archives/2018-11/home/': deleted 43381 directories
2020/10/16 13:35:00 ERROR : Attempt 2/3 succeeded
2020/10/16 13:35:00 INFO  : 
Transferred:   	    6.504M / 6.504 MBytes, 100%, 371 Bytes/s, ETA 0s
Checks:            311611 / 311611, 100%
Deleted:           121115
Transferred:          909 / 909, 100%
Elapsed time:   21h42m3.4s
```

So that's the reason it appeared to be stuck at the end of the run - it was doing the directory deletions, but there was no indication of its progress.

**Comment by ncw** [comment]
OK, so a stat "Deleted dirs:" would work for you ? Maybe I could add it after the Deleted: line so it said something like this?

    Deleted:           121115 (files), 1234 (dirs)

It is annoying when things hang and they don't tell you what they are doing, but it is also annoying if I put too many stats in!

**Comment by gerph** [comment]
That would be really cute; I do understand the hassle of wiring those things up so that they report through to the right place, too.

Whatever works for you. At the moment I'm using -vv all the time because it ensures that I understand that it really is doing something during that end phase, as the deletes turn up there. So you could argue that there's sufficient information if people want it - they just have to up the verbosity.

**Comment by ncw** [comment]
Have a go with this and tell me what you think :-)

[v1.54.0-beta.4849.48e5ff82f.fix-stats](https://beta.rclone.org/branch/fix-stats/v1.54.0-beta.4849.48e5ff82f.fix-stats/) on branch [fix-stats](https://github.com/rclone/rclone/tree/fix-stats) (uploaded in 15-30 mins)

**Comment by gerph** [comment]
Kicked off a sync... I'll let you know how it goes, ta!

**Comment by gerph** [comment]
Yup; I've just been watching some of the transfers at the end, and I can see that it is doing things.

I'm now getting INFO reports that look like this, and it makes it a lot clearer:

```
2019-W40: 2020/10/29 11:11:45 INFO  : 
2019-W40: Transferred:   	    7.180M / 7.180 MBytes, 100%, 16.409 kBytes/s, ETA 0s
2019-W40: Checks:             56394 / 56394, 100%
2019-W40: Deleted:              753 (files), 276 (dirs)
2019-W40: Transferred:         1090 / 1090, 100%
2019-W40: Elapsed time:      9m13.2s
2019-W40: 
2019-W40: 2020/10/29 11:12:18 INFO  : 
2019-W40: Transferred:   	    7.180M / 7.180 MBytes, 100%, 16.409 kBytes/s, ETA 0s
2019-W40: Checks:             56394 / 56394, 100%
2019-W40: Deleted:              753 (files), 376 (dirs)
2019-W40: Transferred:         1090 / 1090, 100%
2019-W40: Elapsed time:      9m46.8s
2019-W40: 
```

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.54 :-)
